sdk/JavaScript/phonegap/www/js/MobileServiceMemoryStore.js

```diff
@@ -0,0 +1,52 @@
+﻿// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ----------------------------------------------------------------------------
+
+zumo.tests.MobileServiceMemoryStore = function () {
+    /// <summary>
+    /// Initializes a new instance of the MobileServiceMemoryStore class.
+    /// </summary>
+
+    var idProperty = "id";
+    var tables = {};
+
+    this.upsert = function (tableName, instance) {

```

Comment:
This raises a good question, I dont think we can have our stores using any promise layer, perhaps this needs to be only callback based.